server/src/CompDBCache.cpp

```diff
@@ -0,0 +1,63 @@
+#include "CompDBCache.h"
+
+#include <sys/stat.h>
+
+#include <cassert>
+
+CompDBCache::CompDBCache()
+  : db_(nullptr), mtime_(0) {
+}
+
+CompDBCache::~CompDBCache() {
+  clear();
+}
+
+CXCompilationDatabase CompDBCache::fromDirectory(const std::string &buildDir,
+                                                 CXCompilationDatabase_Error *error) {
+  assert(error != nullptr);
+
+  const std::string jsonFilename = buildDir + "compile_commands.json";
```

Comment:
Not per se I think, but I couldn't make it (the code as it was before) work if `buildDir` didn't end in a directory separator.

`clang_CompilationDatabase_fromDirectory()` required it. It would not find/load the database without it. That's using libclang that ships with Xcode on macOS though. I assumed it would be the same for all platforms.

Reply:
Oh, I find this very surprising from libclang. :-/

Reply:
Indeed. I can add a check for this though, so that this code will "do the right thing" regardless.